main.tf

```diff
@@ -13,6 +13,15 @@ locals {
   default_ocp_version = "${data.ibm_container_cluster_versions.cluster_versions.default_openshift_version}_openshift"
   ocp_version         = var.ocp_version == null || var.ocp_version == "default" ? local.default_ocp_version : "${var.ocp_version}_openshift"
 
+  # To verify rhcos operating system only for OCP versions >=4.15
```

Comment:
I see we will now have operating_system in both main cluster block, and again in the worker pool blocks. Is there a requirement that the same value be used in both? If so that would be another thing to validate.

Also what do we want to have happen if the main operating_system = "RHCOS" but the worker_pool property is unset (null). Will this cause issues? if worker pool os = null do we need to make sure it matches the cluster setting?

I don't know the answer, just mentioning as something to check.

Reply:
Thank you @toddgiguere for your inputs.

(a) There is no requirement as such to have the same values to be used in both but while creating clusters, we can see one information i.e. 
<img width="797" alt="image" src="https://github.com/terraform-ibm-modules/terraform-ibm-base-ocp-vpc/assets/78964991/bcbdf02b-59c3-4e72-8306-cadd716269c0">

***Choosing RHEL for the default worker pool will limit all additional worker pools to RHEL.If you plan to use RHCOS with this cluster, create the default worker pool with RHCOS***
This statement has confused me and I am thinking to provide a validation like below:

```hcl
# Validate if default worker pool's operating system is RHEL, all pools' operating system must be RHEL
  valid_worker_pools = var.operating_system == "REDHAT_8_64" && alltrue([for worker in var.worker_pools: worker.operating_system == "REDHAT_8_64" ? true: false])
  rhel8_validation = local.valid_worker_pools == true? true: tobool("Choosing RHEL for the default worker pool will limit all additional worker pools to RHEL.")

  # Validate if RHCOS is used as operating system for the cluster then the default worker pool must be created with RHCOS
  rhcos_worker_pools = var.operating_system == "RHCOS" && alltrue([for worker in var.worker_pools: worker.operating_system == "RHCOS" ? true: false])
  default_wp_validation = local.rhcos_worker_pools == true? true: tobool("If you plan to use RHCOS with this cluster, create the default worker pool with RHCOS.")
```


(b) **Another thing regarding validation of `operating_system` at two places is:**
I have added the validation for both cases. Please have a look at these once-<br> 1. `operating_system` not inside worker pool i.e cluster os -  can be seen [here](https://github.com/terraform-ibm-modules/terraform-ibm-base-ocp-vpc/blob/ecc9c3cf4fc38e47d02aabfb5e1da9a9c24c95cd/main.tf#L18)<br>2. `operating_system` within worker pools map - can be seen [here](https://github.com/terraform-ibm-modules/terraform-ibm-base-ocp-vpc/blob/ecc9c3cf4fc38e47d02aabfb5e1da9a9c24c95cd/main.tf#L23)

(c) I am checking this part regarding `null` value of worker pool